lib/engines/content_block_manager/app/models/content_block_manager/content_block/document/document_filter.rb

```diff
@@ -1,15 +1,39 @@
 module ContentBlockManager
   class ContentBlock::Document::DocumentFilter
+    FILTER_ERROR = Data.define(:attribute, :full_message)
+    attr_reader :errors
+
     def initialize(filters = {})
       @filters = filters
+      @errors = []
     end
 
     def paginated_documents
       unpaginated_documents.page(page).per(default_page_size)
     end
 
+    def valid?
+      return @valid if defined?(@valid)
+
+      @valid ||= begin
+        validate_date(:last_updated_from)
+        validate_date(:last_updated_to)
+
+        errors.empty?
+      end
+    end
+
   private
 
+    def validate_date(key)
+      return unless is_date_present?(key)
+
+      date = date_from_filters(key)
+      Time.zone.local(date[:year], date[:month], date[:day])
```

Comment:
This should error if there are missing values right? It's just the commit message seems to suggest otherwise